src/main/java/gregtech/common/metatileentities/multi/electric/generator/MetaTileEntityLightningRod.java

```diff
@@ -0,0 +1,197 @@
+package gregtech.common.metatileentities.multi.electric.generator;
+
+import codechicken.lib.render.CCRenderState;
+import codechicken.lib.render.pipeline.ColourMultiplier;
+import codechicken.lib.render.pipeline.IVertexOperation;
+import codechicken.lib.vec.Matrix4;
+import gregtech.api.GTValues;
+import gregtech.api.capability.impl.EnergyContainerHandler;
+import gregtech.api.gui.ModularUI;
+import gregtech.api.metatileentity.MetaTileEntity;
+import gregtech.api.metatileentity.TieredMetaTileEntity;
+import gregtech.api.metatileentity.interfaces.IGregTechTileEntity;
+import gregtech.api.metatileentity.multiblock.IMultiblockPart;
+import gregtech.api.util.GTUtility;
+import gregtech.api.util.XSTR;
+import gregtech.client.renderer.ICubeRenderer;
+import gregtech.client.renderer.texture.Textures;
+import gregtech.client.renderer.texture.cube.SimpleOverlayRenderer;
+import gregtech.client.renderer.texture.cube.SimpleSidedCubeRenderer;
+import gregtech.client.utils.PipelineUtil;
+import net.minecraft.client.renderer.texture.TextureAtlasSprite;
+import net.minecraft.client.resources.I18n;
+import net.minecraft.entity.effect.EntityLightningBolt;
+import net.minecraft.entity.player.EntityPlayer;
+import net.minecraft.init.Blocks;
+import net.minecraft.item.ItemStack;
+import net.minecraft.network.PacketBuffer;
+import net.minecraft.util.EnumFacing;
+import net.minecraft.util.ResourceLocation;
+import net.minecraft.util.math.BlockPos;
+import net.minecraft.world.World;
+import net.minecraftforge.fml.relauncher.Side;
+import net.minecraftforge.fml.relauncher.SideOnly;
+import org.apache.commons.lang3.ArrayUtils;
+import org.apache.commons.lang3.tuple.Pair;
+
+import javax.annotation.Nonnull;
+import javax.annotation.Nullable;
+import java.util.List;
+
+import static gregtech.api.capability.GregtechDataCodes.IS_WORKING;
+
+public class MetaTileEntityLightningRod extends TieredMetaTileEntity {
+    private boolean isActive = false;
+
+    public MetaTileEntityLightningRod(ResourceLocation metaTileEntityId, int tier) {
+        super(metaTileEntityId, tier);
+    }
+
+    protected void reinitializeEnergyContainer() {
+        if (isEnergyEmitter())
+        { this.energyContainer = EnergyContainerHandler.emitterContainer(this, getEnergyCapacity(), GTValues.V[getTier()], getMaxInputOutputAmperage()); }
+    }
+
+    @Override
+    public void update() {
+        super.update();
+        if(this.getWorld().isRemote)
+            return;
+
+        World aWorld = this.getWorld();
+        XSTR aXSTR = new XSTR();
+        if (!aWorld.isRemote) {
+            if(this.energyContainer.getEnergyStored()>0){
+                this.setActive(true);
+                this.energyContainer.removeEnergy(this.energyContainer.getEnergyStored()/100+1);
+            }else {
+                this.setActive(false);
+            }
+
+            if (getOffsetTimer() % 256 == 0 && (aWorld.isThundering() || (aWorld.isRaining() && aXSTR.nextInt(10) == 0))) {
```

Comment:
I would change this into a base chance of 0 and increase it if its raining or thundering. And then get a random value each second or so.

Reply:
good idea